### PR TITLE
Add interface generator

### PR DESCRIPTION
**Pull request by alex35mil**

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -63,6 +63,13 @@ export default {
         title: 'bsb',
         description: 'The path to the `bsb` binary.',
         default: 'bsb',
+        order: 20,
+      },
+      bsc: {
+        type: 'string',
+        title: 'bsc',
+        description: 'The path to the `bsc` binary.',
+        default: 'bsc',
```
Comment (alex35mil) [inline src/config.ts:72]: I decided to place it next to `bsb` in config, tho it caused some inconvenience w/ types (OSL config doesn't have `bsc` property).

File: src/main.ts [hunk 2]
```diff
@@ -2,17 +2,22 @@ import * as cp from 'child_process'
...
+import { DeepPartial, Config, FileExtension } from './types'
 
 const confFile = ".atom/ide-reason.json"
-const defaultConfig = languageServer.ISettings.defaults.reason
+const defaultConfig = merge(
+  {},
+  languageServer.ISettings.defaults.reason,
+  // @ts-ignore: `bsc` doesn't exist in languageServer.ISettings
+  { path: { bsc: 'bsc' } },
+)
```
Comment (alex35mil) [inline src/main.ts]: As `bsc` doesn't exist in `languageServer.ISettings` I had to ignore this one. Any suggestions how to handle it better are appreciated.

**Comment by alex35mil** [comment]
@zaaack Ready for review.

File: src/main.ts (hunk 2 above)
Comment (zaaack) [reply src/main.ts]: I think you can extends type `Config` with custom fields like this:

```ts
// types.ts
export type OLSConfig = ISettings['reason']
export interface Config extends OLSConfig {
    path: OLSConfig['path'] & { bsc: string }
}
```

then manually pass the generic type to `deepmerge`, thanks for @clouds56 's link for [`RecursivePartial`](https://stackoverflow.com/questions/41980195/recursive-partialt-in-typescript-2-1), it should handle this case perfectly:

```ts
// main.ts
const defaultConfig = merge<Config>(
  languageServer.ISettings.defaults.reason,
  {
    path: {
      bsb: 'bsb'
    }
  }
)
```

ps:`deepmerge` can only merge two objects one time, the third parameter is options.

Comment (alex35mil) [reply src/main.ts]: Works like a charm, thanks!

**Comment by zaaack** [comment]
Just published ide-reason@v0.1.13!  🎉 🎉 

Sorry that I didn't notice your todo list in the readme, I simply implemented [namespace support](https://github.com/reasonml-editor/atom-ide-reason/commit/51a8513da7b5ac4bb4526e4782613bafaff13f96) when trying locally.

**Comment by alex35mil** [comment]
Awesome, thanks!

**Pull request merged**